src/client/ua_client_connect.c

```diff
@@ -394,27 +394,29 @@ decodeProcessOPNResponseAsync(void *application, UA_SecureChannel *channel,
         return;
     }
 
-    UA_UInt32 sequenceNumber = 0;
     retval = decryptAndVerifyChunk(channel, &channel->securityPolicy->asymmetricModule.cryptoModule,
-                                   UA_MESSAGETYPE_OPN, msg, offset, &requestId, &sequenceNumber);
+                                   UA_MESSAGETYPE_OPN, msg, offset);
     if(retval != UA_STATUSCODE_GOOD) {
         UA_LOG_WARNING_CHANNEL(&client->config.logger, channel,
                                "Could not decrypt and verify the OPN payload");
         UA_Client_disconnect(client);
         return;
     }
 
+   /* Decode and verify the sequence header */
+    UA_SequenceHeader sequenceHeader;
+    retval = UA_SequenceHeader_decodeBinary(msg, &offset, &sequenceHeader);
 #ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
-    retval = processSequenceNumberAsym(channel, sequenceNumber);
+    retval |= processSequenceNumberAsym(channel, sequenceHeader.sequenceNumber);
```

Comment:
OR'ing the status codes is a bad idea. If two bad status codes are OR'ed it is impossible to reconstruct both error codes in most cases.